### PR TITLE
Improve the translation of file names to module names.

### PR DESCRIPTION
**Pull request by ciechowoj**
As in the title.

### CODE REVIEW

**Comment by ciechowoj** [comment]
I needed to bump the DMD version for appveyor.
For 2.071.0 it misses some functions in std.uni (e.g. isAlphaNum).
For 2.072.0 it did't want to compile with x86_mscoff.
I bumped it to 2.073.0, however I needed to make some modifications to `unit_test/HttpClient.d`.

**Comment by jacob-carlborg** [comment]
> I needed to bump the DMD version for appveyor.

Feel free to bump to the latest version. The Travis CI tests are running the latest version of DMD and LDC.

**Comment by jacob-carlborg** [comment]
1. I would prefer that this was implemented as an option. It don't want to take to much freedom in the translation, DStep still needs to be a generic tool
1. I would prefer that it translated camel case to snake case, i.e. translate `CXCompilationDatabase` to `cx_compilation_database` instead of `cxcompilationdatabase`

**Comment by ciechowoj** [comment]
> Feel free to bump to the latest version. The Travis CI tests are running the latest version of DMD and LDC.

Bumped to  2.076.0.

> I would prefer that this was implemented as an option. It don't want to take to much freedom in the translation, DStep still needs to be a generic tool

Done, however I left replacing unsupported characters (`@`, `#`) with `.`.

> I would prefer that it translated camel case to snake case, i.e. translate CXCompilationDatabase to cx_compilation_database instead of cxcompilationdatabase

Done.

File: dstep/Configuration.d [hunk 1]
```diff
@@ -49,6 +49,10 @@ struct Configuration
     @("public-submodules", "Use public imports for submodules [default].")
     bool publicSubmodules = false;
 
+    // rename imports of submodules to D conforming form
+    @("normalize-submodules", "Rename imports of submodules to D conforming form [default].")
```
Comment (jacob-carlborg) [inline dstep/Configuration.d]: What is `submodules` referring to? Shouldn't it be `module`?

Comment (ciechowoj) [reply dstep/Configuration.d]: `normalize-modules`?
`rename-module-names`?
`rename-modules`?

Comment (jacob-carlborg) [reply dstep/Configuration.d]: `normalize-modules` sounds better.

**Comment by ciechowoj** [comment]
Ups, I broke this PR do not merge...

**Comment by ciechowoj** [comment]
Fixed.

However for some reason unix tests are failing with:
`The command "source "$(CURL_USER_AGENT="$CURL_USER_AGENT" bash install.sh dmd --activate)"" failed and exited with 1 during .`

**Comment by ciechowoj** [comment]
Not only in this branch...

**Comment by ZombineDev** [comment]
The same error happened on other D projects and it's due to some intermittent network problem. Hopefully it will go away soon enough.

File: appveyor.yml [hunk 2]
```diff
@@ -8,8 +8,13 @@ configuration:
...
 environment:
-  DVersion: 2.071.0
-  DC: dmd
+  matrix:
+  - DC: dmd
+    DVersion: 2.076.0
+    arch: x86
+  - DC: dmd
+    DVersion: 2.076.0
+    arch: x86_64
```
Comment (jacob-carlborg) [inline appveyor.yml]: I've removed the usage of `arch`. The appveyor script now uses the `PLATFORM` environment variable to identify the architecture. A build matrix is automatically created when `platform` [1] is used, which will automatically set the `PLATFORM` environment variable.

[1] https://github.com/jacob-carlborg/dstep/blob/master/appveyor.yml#L1

Comment (ciechowoj) [reply appveyor.yml]: Fixed.

**Comment by jacob-carlborg** [comment]
Please add a changelog entry.

**Comment by jacob-carlborg** [comment]
👌

**Comment by jacob-carlborg** [comment]
Thanks.

**Pull request merged**